File: app/src/main/java/com/demo/assignment/repository/model/LoginModel.java
package com.demo.assignment.repository.model;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;

import com.demo.assignment.BR;

public class LoginModel extends BaseObservable {
    private String firstName;
    private String lastName;
    private boolean isSubmitEnable;

    @Bindable
    public String getFirstName() {
        return this.firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
        notifyPropertyChanged(BR.firstName);
    }

    @Bindable
    public String getLastName() {
        return this.lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
        notifyPropertyChanged(BR.lastName);
    }

    @Bindable
    public boolean isSubmitEnable() {
        return isSubmitEnable;
    }

    public void setSubmitEnable(boolean submitEnable) {
        isSubmitEnable = submitEnable;
        notifyPropertyChanged(BR.submitEnable);
    }
}

File: app/src/main/java/com/demo/assignment/repository/model/LoginModel.kt
package com.demo.assignment.repository.model

import android.text.TextUtils
import androidx.databinding.BaseObservable
import androidx.databinding.Bindable
import com.demo.assignment.BR

class LoginModel : BaseObservable() {

    @get:Bindable
    var firstName: String = ""
        set(value) {
            field = value
            notifyPropertyChanged(BR.firstName)
        }

    @get:Bindable
    var lastName: String = ""
        set(value) {
            field = value
            notifyPropertyChanged(BR.lastName)
        }


    @get:Bindable
    var isSubmitEnable: Boolean = false
        set(value) {
            field = value
            notifyPropertyChanged(BR.submitEnable)
        }

    @Suppress("UNUSED_PARAMETER")
    fun onTextChanged(text: CharSequence, start: Int, before: Int,
                      count: Int) {
        isSubmitEnable = (!TextUtils.isEmpty(firstName)
                && !TextUtils.isEmpty(text)
                && !TextUtils.isEmpty(lastName))
    }


}